### DISCUSSION TITLE
Open Ai model

### DISCUSSION THREAD

**Discussion by popov-ig** [IliaLarchenko/Interviewer#1] [General]
Hi @IliaLarchenko!

Thanks for creating your AI Interviewer. It looks really interesting!

I noticed you're using the OpenAI Whisper API instead of running it locally. 
Could you share your reasoning behind this decision? I'm curious to understand the trade-offs you considered.

Specifically, I saw [this line in the audio.py file](https://github.com/IliaLarchenko/Interviewer/blob/main/api/audio.py#L110C1-L116C18). 
Would you mind elaborating on why you chose to go with the API approach?

I'm looking forward to learning more about your project!

**Comment by IliaLarchenko** [comment]
Hi @popov-ig !

It is a good question. 
You can run Whisper locally if you use the following setting in the `.env` file.
```
STT_URL=None
STT_TYPE=HF_LOCAL
STT_NAME=openai/whisper-base.en
```
You can choose any model, I recommend a Large one if your hardware is fast enough.

The main consideration from my side was response time. I recently tested [different ways to run Whisper](https://www.linkedin.com/posts/larchenko_whisper-inference-benchmark-activity-7200837606979354624-IDms). 

A simple conclusion is if you use it on a machine with a powerful GPU you can run it locally and it will be the fastest option. If not - OpenAI API is the fastest one.

I usually run it from my laptop or using HF spaces (without GPU), so my default choice is OpenAI API.
